### PR TITLE
Integration testing

### PR DESCRIPTION
**Pull request by emmanuelsteven**
In this PR I included integration specs for every feature implemented!
before
after

- User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
-  user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
- User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
- Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

File: spec/models/pot_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Pot, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (ShadyShawkat) [inline spec/models/pot_spec.rb]: - [ ] Kindly fix the following error that prevented all of the tests from running.

image

File: spec/features/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      expect(page).to have_content(@third_post.text)
+    end
+    it 'I can see a button that lets me view all of a user\'s posts.' do
+      expect(page).to have_selector("a[href='#{user_posts_path(@first_user)}']")
+    end
+    it "When I click a user's post, it redirects me to that post's show page." do
+      find("a[href='#{user_post_path(@first_user, @first_post)}']").click
+      expect(page).to have_current_path(user_post_path(@first_user, @first_post))
+    end
+    it "When I click to see all posts, it redirects me to the user's post's index page." do
```
Comment (ShadyShawkat) [inline spec/features/user_show_spec.rb:45]: - [ ] When I tried to run tests, the following tests failed, kindly fix them. If you want to replicate the errors I had, please don't forget to drop and recreate the database and then try again.

image

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @emmanuelsteven, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/pot_spec.rb (hunk 1 above)
Comment (emmanuelsteven) [reply spec/models/pot_spec.rb]: thank you for spotting this out It's an error file I forgot to delete it. however, I have deleted it now as it doesn't contribute to an aspect of my app.

File: spec/features/user_show_spec.rb (hunk 2 above)
Comment (emmanuelsteven) [reply spec/features/user_show_spec.rb:45]: every bit of my code worked as required, I think you just might be entering the wrong directory, my test files aren't in the request directory but in spec/features.

**Review by cynthiainga — APPROVED** [review]
Hi @emmanuelsteven, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] The integration tests are passing ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] Descriptive PR ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**